src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -71,92 +71,148 @@ private class ConfigurationCacheItem
 
         public Task<bool> IsEnabledAsync(string feature)
         {
-            return IsEnabledWithVariantsAsync<object>(feature, appContext: null, useAppContext: false, CancellationToken.None).AsTask();
+            return IsEnabledEvaluation<object>(feature, appContext: null, useAppContext: false, CancellationToken.None).AsTask();
         }
 
         public Task<bool> IsEnabledAsync<TContext>(string feature, TContext appContext)
         {
-            return IsEnabledWithVariantsAsync(feature, appContext, useAppContext: true, CancellationToken.None).AsTask();
+            return IsEnabledEvaluation(feature, appContext, useAppContext: true, CancellationToken.None).AsTask();
         }
 
         public ValueTask<bool> IsEnabledAsync(string feature, CancellationToken cancellationToken)
         {
-            return IsEnabledWithVariantsAsync<object>(feature, appContext: null, useAppContext: false, cancellationToken);
+            return IsEnabledEvaluation<object>(feature, appContext: null, useAppContext: false, cancellationToken);
         }
 
         public ValueTask<bool> IsEnabledAsync<TContext>(string feature, TContext appContext, CancellationToken cancellationToken)
         {
-            return IsEnabledWithVariantsAsync(feature, appContext, useAppContext: true, cancellationToken);
+            return IsEnabledEvaluation(feature, appContext, useAppContext: true, cancellationToken);
         }
 
-        private async ValueTask<bool> IsEnabledWithVariantsAsync<TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
+        private async ValueTask<bool> IsEnabledEvaluation<TContext>(string feature, TContext appContext, bool useAppContext, CancellationToken cancellationToken)
```

Comment:
According to https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-type-members#names-of-methods
The method name should be verb. Is the name "IsEnabledEvaluation" a noun?

I suggest "EvaluateFeature".

Reply:
We may want to avoid this change given IFeatureManager's main method is `IsEnabled`. That's not a verb. I don't want to stray to far from that with the internal methods.

Reply:
You raise a good point, perhaps we could have named `IsEnabled` something like `GetXyz` in the beginning.